app/src/androidTest/java/com/google/samples/apps/topeka/activity/SignInActivityTest.java

```diff
@@ -65,20 +72,142 @@ public void checkFab_initiallyNotDisplayed() {
+                .check(matches(isDisplayed()))
+                .perform(click()).check(matches(isChecked()));
+    }
+
+    @Test
+    public void signIn_chooseAvatar_08() {
+        onView(allOf(withParent(withId(R.id.avatars)), withContentDescription("Avatar 71")))
+                .check(matches(isDisplayed()))
+                .perform(click()).check(matches(isChecked()));
+    }
+
+    @Test
+    public void signIn_chooseAvatar_09() {
+        onView(allOf(withParent(withId(R.id.avatars)), withContentDescription("Avatar 81")))
+                .check(matches(isDisplayed()))
+                .perform(click()).check(matches(isChecked()));
+    }
+
+    @Test
+    public void signIn_chooseAvatar_10() {
+        onView(allOf(withParent(withId(R.id.avatars)), withContentDescription("Avatar 91")))
+                .check(matches(isDisplayed()))
+                .perform(click()).check(matches(isChecked()));
+    }
+
+    @Test
+    public void signIn_chooseAvatar_11() {
+        onView(allOf(withParent(withId(R.id.avatars)), withContentDescription("Avatar 101")))
+                .check(matches(isDisplayed()))
+                .perform(click()).check(matches(isChecked()));
+    }
+
+    @Test
+    public void signIn_chooseAvatar_12() {
+        onView(allOf(withParent(withId(R.id.avatars)), withContentDescription("Avatar 111")))
+                .check(matches(isDisplayed()))
+                .perform(click()).check(matches(isChecked()));
+    }
+
+    @Test
+    public void signIn_chooseAvatar_13() {
+        onView(allOf(withParent(withId(R.id.avatars)), withContentDescription("Avatar 121")))
+                .check(matches(isDisplayed()))
+                .perform(click()).check(matches(isChecked()));
+    }
+
+    @Test
+    public void signIn_chooseAvatar_14() {
+        onView(allOf(withParent(withId(R.id.avatars)), withContentDescription("Avatar 131")))
+                .check(matches(isDisplayed()))
+                .perform(click()).check(matches(isChecked()));
+    }
+
+    @Test
+    public void signIn_chooseAvatar_15() {
+        onView(allOf(withParent(withId(R.id.avatars)), withContentDescription("Avatar 141")))
+                .check(matches(isDisplayed()))
+                .perform(click()).check(matches(isChecked()));
+    }
+
+    @Test
+    public void signIn_chooseAvatar_16() {
+        onView(allOf(withParent(withId(R.id.avatars)), withContentDescription("Avatar 151")))
+                .check(matches(isDisplayed()))
+                .perform(click()).check(matches(isChecked()));
     }
 
     @Test
     public void signIn_withLongLastName() {
         inputData();
-        onView(withId(R.id.last_initial)).perform(typeText("somelongtext"), closeSoftKeyboard());
-        onView(withId(R.id.last_initial)).check(matches(withText(TEST_LAST_INITIAL)));
+        onView(withId(last_initial)).perform(typeText(InstrumentationRegistry.getTargetContext().getResources().getString(R.string.last_initial)), closeSoftKeyboard());
```

Comment:
Also, I just saw that you could just use `TEST_FIRST_NAME` instead of loading a new resource, which fulfills the same purpose.